pkg/prometheus/resource_selector.go

```diff
@@ -1590,9 +1664,31 @@ func (rs *ResourceSelector) validateIonosSDConfigs(ctx context.Context, sc *moni
 			return fmt.Errorf("[%d]: %w", i, err)
 		}
 
+		if err := validateCustomHTTPConfig(ctx, config.CustomHTTPConfig, rs.store, sc.GetNamespace()); err != nil {
+			return fmt.Errorf("[%d]: %w", i, err)
+		}
+
 		if err := rs.store.AddSafeTLSConfig(ctx, sc.GetNamespace(), config.TLSConfig); err != nil {
 			return fmt.Errorf("[%d]: %w", i, err)
 		}
 	}
 	return nil
 }
+
+func validateCustomHTTPConfig(ctx context.Context, hh monitoringv1.CustomHTTPConfig, store *assets.StoreBuilder, namespace string) error {
+	if reflect.ValueOf(hh).IsZero() || len(hh.HTTPHeaders) <= 0 {
+		return nil
+	}
+
+	for k, v := range hh.HTTPHeaders {
```

Comment:
Maybe worth to check version here as well and reject resource.

Reply:
Same